src/GraphQL.ApiTests/GraphQL.approved.txt

```diff
@@ -2725,11 +2743,17 @@ namespace GraphQL.Validation
     {
         public static readonly System.Collections.Generic.IEnumerable<GraphQL.Validation.IValidationRule> CoreRules;
         public DocumentValidator() { }
-        public System.Threading.Tasks.Task<GraphQL.Validation.IValidationResult> ValidateAsync(string originalQuery, GraphQL.Types.ISchema schema, GraphQL.Language.AST.Document document, System.Collections.Generic.IEnumerable<GraphQL.Validation.IValidationRule> rules = null, System.Collections.Generic.IDictionary<string, object> userContext = null, GraphQL.Inputs inputs = null) { }
+        [return: System.Runtime.CompilerServices.TupleElementNames(new string[] {
+                "validationResult",
+                "variables"})]
+        public System.Threading.Tasks.Task<System.ValueTuple<GraphQL.Validation.IValidationResult, GraphQL.Language.AST.Variables>> ValidateAsync(string originalQuery, GraphQL.Types.ISchema schema, GraphQL.Language.AST.Document document, GraphQL.Language.AST.Operation operation, System.Collections.Generic.IEnumerable<GraphQL.Validation.IValidationRule> rules = null, System.Collections.Generic.IDictionary<string, object> userContext = null, GraphQL.Inputs inputs = null) { }
```

Comment:
I looked through the rest of the code.  It seems the operation is only used to retrieve the `Variables` from it.  I think we should change the signature of `ValidateAsync` to include the variables directly and not include the `Operation`.  It will be clearer that we are validating given a set of variables, and less likely for any future changes to the context to accidentally be based on the selected operation.

Reply:
> Adding the operation to the validation context can break document caching

How?

> to include the variables directly and not include the Operation

Reply:
Let's say `Operation` is available for validation rules to operate against.  Let's assume there is a simple custom validation rule: "if operation type is query then pass else fail".  When document caching is used, and if a document with a query operation (no variables) is validated, the document will be cached.  If the same document is executed again with a mutation operation (no variables), the document will be pulled from the cache and assumed to be valid.

Now I understand:
1. Currently the `Operation` is not cached in the validation context and hence not accessible by a validation rule (although it is available by a `IDocumentValidator` implementation).
2. The rule can be added to the `CachedDocumentValidationRules` to alleviate this issue.

However, the purpose of `Operation` being passed is simply to retrieve the variables, so I suggested that it would be better to do that prior to the `DocumentValidator` execution.

Reply:
>  it would be better to do that prior to the DocumentValidator execution.

Reply:
There is a chicken and egg problem.

Reply:
I just meant to pass the variables directly in. operation.Variables instead of operation

Reply:
Oooooooops 👍

Reply:
Done.